File: src/main/java/com/example/demoAllTouhou/services/MercancyService.java
package com.example.demoAllTouhou.services;

import com.example.demoAllTouhou.entities.MercancyEntity;
import com.example.demoAllTouhou.repositories.MercancyRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class MercancyService {

    @Autowired
    private MercancyRepository mercancyRepository;

    // **Crear una nueva Mercancía**
    public MercancyEntity createMercancy(MercancyEntity mercancyEntity) {
        return mercancyRepository.save(mercancyEntity);
    }

    // **Obtener todas las Mercancías**
    public List<MercancyEntity> getAllMercancy() {
        return mercancyRepository.findAll();
    }

    // **Obtener Mercancía por ID**
    public MercancyEntity getMercancyById(Long id) {
        return mercancyRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Mercancy not found with id: " + id));
    }

    // **Actualizar una Mercancía**
    public MercancyEntity updateMercancy(Long id, MercancyEntity updatedMercancy) {
        MercancyEntity existingMercancy = mercancyRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Mercancy not found with id: " + id));

        existingMercancy.setCharacterId(updatedMercancy.getCharacterId());
        existingMercancy.setName(updatedMercancy.getName());
        existingMercancy.setDescription(updatedMercancy.getDescription());
        existingMercancy.setPrice(updatedMercancy.getPrice());
        existingMercancy.setUpdateDate(updatedMercancy.getUpdateDate());
        existingMercancy.setLink(updatedMercancy.getLink());

        return mercancyRepository.save(existingMercancy);
    }

    // **Eliminar una Mercancía por ID**
    public void deleteMercancy(Long id) {
        if (!mercancyRepository.existsById(id)) {
            throw new RuntimeException("Mercancy not found with id: " + id);
        }
        mercancyRepository.deleteById(id);
    }

    // **Obtener Mercancías por Character ID**
    public List<MercancyEntity> getMercancyByCharacterId(long characterId) {
        return mercancyRepository.findByCharacterId(characterId);
    }

    // **Obtener Mercancías que contienen un texto en el nombre**
    public List<MercancyEntity> getMercancyByName(String name) {
        return mercancyRepository.findByNameContainingIgnoreCase(name);
    }

    // **Obtener Mercancías con precio superior al ingresado**
    public List<MercancyEntity> getMercancyByPriceGreaterThan(int price) {
        return mercancyRepository.findByPriceGreaterThan(price);
    }

    // **Obtener Mercancías con precio inferior al ingresado**
    public List<MercancyEntity> getMercancyByPriceLessThan(int price) {
        return mercancyRepository.findByPriceLessThan(price);
    }
}

File: Frontend/src/services/MercancyService.js
import httpClient from "../http-common";

// Crea una nueva Mercancía
const createmerch = (merch) => {
    return httpClient.post(`/mercancies`, merch);
};

// Obtiene todas las Mercancías
const getAllmerch = () => {
    return httpClient.get(`/mercancies`);
};

// Obtiene una Mercancía por ID
const getmerchById = (id) => {
    return httpClient.get(`/mercancies/${id}`);
};

// Actualiza una Mercancía
const updatemerch = (id, merch) => {
    return httpClient.put(`/mercancies/${id}`, merch);
};

// Elimina una Mercancía por ID
const deletemerch = (id) => {
    return httpClient.delete(`/mercancies/${id}`);
};

// Obtiene Mercancías por Character ID
const getmerchByCharacterId = (characterId) => {
    return httpClient.get(`/mercancies/by-character-id/${characterId}`);
};

// Obtiene Mercancías que contienen un texto en el nombre
const getmerchByName = (name) => {
    return httpClient.get(`/mercancies/by-name`, { params: { name } });
};

// Obtiene Mercancías con precio superior al ingresado
const getmerchByPriceGreaterThan = (price) => {
    return httpClient.get(`/mercancies/by-price-greater-than/${price}`);
};

// Obtiene Mercancías con precio inferior al ingresado
const getmerchByPriceLessThan = (price) => {
    return httpClient.get(`/mercancies/by-price-less-than/${price}`);
};

export default {
    createmerch,
    getAllmerch,
    getmerchById,
    updatemerch,
    deletemerch,
    getmerchByCharacterId,
    getmerchByName,
    getmerchByPriceGreaterThan,
    getmerchByPriceLessThan
};
